spec/mongo/server/connection_spec.rb

```diff
@@ -74,16 +100,16 @@ class ConnectionSpecTestException < Exception; end
 
       shared_examples_for 'failing connection' do
         it 'raises an exception' do
-          expect(result).to be_a(Exception)
+          expect(error).to be_a(Exception)
         end
 
         it 'clears socket' do
-          result
+          error
           expect(connection.send(:socket)).to be nil
         end
 
         it 'attempts to reconnect after failure when asked' do
-          # for some reason referencing result here instead of
+          # for some reason referencing error here instead of
```

Comment:
I think it might be due to laziness? I think the `let` block is only evaluated once and then uses the previous result for subsequent evaluations. Since `authenticate!` was already called in the tests above, it isn't called this time.

Reply:
I think I only used it once but maybe not, I'll keep this in mind for the future.